panic: runtime error: invalid memory address or nil pointer dereference

```
root@host:~# cgrps ls
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x55b1eb]

goroutine 1 [running]:
github.com/k1LoW/cgrps/cgroups.(*Cgroups).List.func1(0xc4201b57c0, 0x3d, 0x0, 0x0, 0x756960, 0xc4203f5500, 0x0, 0x0)
        /Users/k1low/src/github.com/k1LoW/cgrps/cgroups/cgroups.go:81 +0x5b
path/filepath.walk(0xc4201c2860, 0x13, 0x759cc0, 0xc4201cf380, 0xc4201cc5d0, 0x0, 0x30)
        /Users/k1low/.anyenv/envs/goenv/versions/1.8.3/src/path/filepath/path.go:372 +0x2fe
path/filepath.Walk(0xc4201c2860, 0x13, 0xc4201cc5d0, 0x13, 0x2)
        /Users/k1low/.anyenv/envs/goenv/versions/1.8.3/src/path/filepath/path.go:398 +0x14c
github.com/k1LoW/cgrps/cgroups.(*Cgroups).List(0xc42006fbc8, 0x659f64, 0x32, 0x791280, 0xc4200ae800, 0x0)
        /Users/k1low/src/github.com/k1LoW/cgrps/cgroups/cgroups.go:89 +0x2f1
github.com/k1LoW/cgrps/cmd.glob..func2(0x771ce0, 0x790840, 0x0, 0x0)
        /Users/k1low/src/github.com/k1LoW/cgrps/cmd/ls.go:39 +0x76
github.com/k1LoW/cgrps/vendor/github.com/spf13/cobra.(*Command).execute(0x771ce0, 0x790840, 0x0, 0x0, 0x771ce0, 0x790840)
        /Users/k1low/src/github.com/k1LoW/cgrps/vendor/github.com/spf13/cobra/command.go:760 +0x2b9
github.com/k1LoW/cgrps/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x7721a0, 0xc42009e058, 0x0, 0x7905eb)
        /Users/k1low/src/github.com/k1LoW/cgrps/vendor/github.com/spf13/cobra/command.go:846 +0x37e
github.com/k1LoW/cgrps/vendor/github.com/spf13/cobra.(*Command).Execute(0x7721a0, 0x7721a0, 0xc42006ff50)
        /Users/k1low/src/github.com/k1LoW/cgrps/vendor/github.com/spf13/cobra/command.go:794 +0x2b
github.com/k1LoW/cgrps/cmd.Execute()
        /Users/k1low/src/github.com/k1LoW/cgrps/cmd/root.go:46 +0x31
main.main()
        /Users/k1low/src/github.com/k1LoW/cgrps/main.go:20 +0x20
```

--- a/cgroups/cgroups.go
+++ b/cgroups/cgroups.go
@@ -78,6 +78,12 @@ func (c *Cgroups) List() ([]string, error) {
 		searchDir := filepath.Clean(fmt.Sprintf("%s/%s", c.FsPath, s))
 
 		err := filepath.Walk(searchDir, func(path string, f os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if f == nil {
+				return nil
+			}
 			if f.IsDir() {
 				c := strings.Replace(path, searchDir, "", 1)
 				if c != "" && !encountered[c] {

--- a/cgroups/cgroups.go
+++ b/cgroups/cgroups.go
@@ -79,7 +79,7 @@ func (c *Cgroups) List() ([]string, error) {
 
 		err := filepath.Walk(searchDir, func(path string, f os.FileInfo, err error) error {
 			if err != nil {
-				return err
+				// continue walking
 			}
 			if f == nil {
 				return nil

